### PR TITLE
remove hardcoded delegator emissions percentage

### PR DESCRIPTION
**Pull request by ThornM9**

### CODE REVIEW

File: packages/helium-cli/src/create-subdao.ts [hunk 1]
```diff
@@ -181,6 +181,12 @@ const yarg = yargs(hideBin(process.argv)).options({
     describe: "Authority index for squads. Defaults to 1",
     default: 1,
   },
+  delegatorRewardsPercent: {
+    type: "number",
+    required: false,
+    describe: "Percentage of rewards allocated to delegators. Must be between 0-100 and can have 2 decimal places.",
+    default: 6,
```
Comment (bryzettler) [inline packages/helium-cli/src/create-subdao.ts:188]: Do we want this defaulted to 6 or just a required field on the script?

**Review by bryzettler — APPROVED** [review]

File: packages/helium-cli/src/create-subdao.ts [hunk 2]
```diff
@@ -182,6 +183,11 @@ const yarg = yargs(hideBin(process.argv)).options({
     describe: "Authority index for squads. Defaults to 1",
     default: 1,
   },
+  delegatorRewardsPercent: {
+    type: "number",
+    required: true,
+    describe: "Percentage of rewards allocated to delegators. Must be between 0-100 and can have 2 decimal places.",
```
Comment (ChewingGlass) [inline packages/helium-cli/src/create-subdao.ts:189]: This is wrong?

File: packages/helium-cli/src/update-subdao.ts [hunk 3]
```diff
@@ -93,6 +94,12 @@ const yarg = yargs(hideBin(process.argv)).options({
     describe: "VSR Registrar of subdao",
     default: null,
   },
+  delegatorRewardsPercent: {
+    type: "number",
+    required: false,
+    describe: "Percentage of rewards allocated to delegators. Must be between 0-100 and can have 2 decimal places.",
```
Comment (ChewingGlass) [inline packages/helium-cli/src/update-subdao.ts:100]: again, not right

File: programs/helium-sub-daos/src/instructions/initialize_sub_dao_v0.rs [hunk 4]
```diff
@@ -277,6 +278,12 @@ pub fn handler(ctx: Context<InitializeSubDaoV0>, args: InitializeSubDaoArgsV0) -
     Some(ctx.accounts.sub_dao.key()),
   )?;
 
+  assert!(
+    args.delegator_rewards_percent
+      < 100_u64
+        .checked_mul(10_u64.checked_pow(8_u32).unwrap())
```
Comment (ChewingGlass) [inline programs/helium-sub-daos/src/instructions/initialize_sub_dao_v0.rs:284]: burning compute, prefer `10_00000000`

File: programs/helium-sub-daos/src/instructions/initialize_sub_dao_v0.rs [hunk 5]
```diff
@@ -277,6 +278,12 @@ pub fn handler(ctx: Context<InitializeSubDaoV0>, args: InitializeSubDaoArgsV0) -
     Some(ctx.accounts.sub_dao.key()),
   )?;
 
+  assert!(
```
Comment (ChewingGlass) [inline programs/helium-sub-daos/src/instructions/initialize_sub_dao_v0.rs:281]: `require_lt!` will give a better error

File: programs/helium-sub-daos/src/instructions/issue_rewards_v0.rs [hunk 6]
```diff
@@ -170,6 +170,9 @@ pub fn handler(ctx: Context<IssueRewardsV0>, args: IssueRewardsArgsV0) -> Result
     .unwrap();
 
   let delegators_present = ctx.accounts.sub_dao_epoch_info.vehnt_at_epoch_start > 0;
+  let max_percent = 100_u64
+    .checked_mul(10_u64.checked_pow(8_u32).unwrap())
```
Comment (ChewingGlass) [inline programs/helium-sub-daos/src/instructions/issue_rewards_v0.rs:174]: same as above

File: programs/helium-sub-daos/src/instructions/issue_rewards_v0.rs [hunk 7]
```diff
@@ -178,33 +181,35 @@ pub fn handler(ctx: Context<IssueRewardsV0>, args: IssueRewardsArgsV0) -> Result
...
     },
   )?;
 
   let delegation_rewards_amount = if delegators_present {
     total_emissions
-      .checked_mul(6)
+      .checked_mul(ctx.accounts.sub_dao.delegator_rewards_percent)
       .unwrap()
-      .checked_div(100)
+      .checked_div(max_percent) // 100% with 2 decimals accuracy
```
Comment (ChewingGlass) [inline programs/helium-sub-daos/src/instructions/issue_rewards_v0.rs:195]: comment is wrong

File: programs/helium-sub-daos/src/instructions/update_sub_dao_v0.rs [hunk 8]
```diff
@@ -53,6 +54,14 @@ pub fn handler(ctx: Context<UpdateSubDaoV0>, args: UpdateSubDaoArgsV0) -> Result
     ctx.accounts.sub_dao.registrar = registrar;
   }
 
+  let max_percent = 100_u64
+    .checked_mul(10_u64.checked_pow(8_u32).unwrap())
+    .unwrap();
+  if let Some(delegator_rewards_percent) = args.delegator_rewards_percent {
+    assert!(delegator_rewards_percent < max_percent);
```
Comment (ChewingGlass) [inline programs/helium-sub-daos/src/instructions/update_sub_dao_v0.rs:61]: same as above

File: tests/utils/daos.ts [hunk 9]
```diff
@@ -124,6 +124,7 @@ export async function initTestSubdao(
         threshold: new anchor.BN("10000000000000000000"),
       },
       dcBurnAuthority: authority,
+      delegatorRewardsPercent: new BN(6 * Math.pow(10, 8)), // 6%
```
Comment (ChewingGlass) [inline tests/utils/daos.ts:127]: Should make a helper function for this. Ex `delegatorRewardsPercent(10.25666)`

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**